Feature/appv1

tnyandoro opened the pull request:
Weather app

tnyandoro committed:
Setup webpack and babel to transpile to older browsers

tnyandoro committed:
Add the html structure and crate the search form

tnyandoro committed:
Add thw stylesheet

tnyandoro committed:
Add styling to the homepage

tnyandoro committed:
Get API to collect weather data

tnyandoro committed:
display weather and city

tnyandoro committed:
Remove the nd

tnyandoro committed:
Add the dom displays

tnyandoro committed:
Add api function

tnyandoro committed:
display weather data

tnyandoro committed:
Add description styling

tnyandoro committed:
Add event listener for the search

tnyandoro committed:
add the search weather function

tnyandoro committed:
Add a listener for the enter key

tnyandoro committed:
Set default weather to my local city

tnyandoro committed:
add matching backgrounds of cities

tnyandoro committed:
add the readme

tnyandoro committed:
add the links of acknowledgements

tnyandoro committed:
Fix webpack

tnyandoro committed:
Fix webpack error

tnyandoro committed:
Merge branch 'feature/appv1' of https://github.com/tnyandoro/weather_app into feature/appv1

tnyandoro committed:
Add the app workflow

tnyandoro committed:
Revert "Add the app workflow"

This reverts commit 4f402d47599d04c81413bb4014646be84edd9e03.

tnyandoro committed:
Create linters.yml

teekaytech reviewed (changes requested):
## Status: Invalid Code Review Request :x: 

__Hi @tnyandoro, I commend you for the great effort put into this project. However, I have to mark the project as invalid due to the following reasons:__

- [ ] Somall linter che checks were not successful: to consider this project ready for review, you need to ensure that all linter checks are running/passing successfully. 
- [ ] Your pull request does not have a meaningful title and a short summary. You need to summarize what you've done in this project in your pull request message.

**Friendly reminder:** You should make sure that your repo looks professional before review.
Please take a step back and read a lesson [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/).
Do not forget to apply 3 simple rules from this lesson (it will be easy as you can use a template provided there). This Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready!

_**Do ensure that the issues above are fixed before requesting another review.**_

* * *
**Please, do not open a new Pull Request for subsequent reviews. You should use the same Pull Request submitted for the first review unless it is requested otherwise.**

**Always post your inquiries and questions on this PR thread as a comment. You can mention me (@teekaytech) so I can see your comments and respond.**

* * * 

**_You can reach out to me on Twitter (@ola_lere), Github(@teekaytech), or slack (@taofeekolalere )_**

Happy Coding :man_technologist:

tnyandoro committed:
linters

tnyandoro committed:
Edit the README markup

tnyandoro committed:
Add fixed code

whiz25 reviewed (changes requested):
## Changes Required :pray:
Nice work on this project. I like the Apple design for the UI. 
I was able to search for weather based on different locations and the background image changed based on the data received. Well done!
Let's make some crucial changes before merging this PR.
- [ ] Add a toggle button for changing temperature unit between Fahrenheit and Celsius
- [ ] Remove comments from `index.js` and `weather.js` files. Comments make your code look ugly.
- [ ] The `index.js` and `weather.js` files both have similar code. Remove one file since it's redundant.
- [ ] Add error handling to the `promises` logic.
- [ ] Update readme instructions. Check my comments for suggestions.
- [ ] Make your PR more descriptive


### Martin Afani
#### Slack: whiz25

tnyandoro committed:
Add changes to the readme

tnyandoro committed:
Add the new live link

tnyandoro committed:
Fix the mode

tnyandoro committed:
Fix promise

tnyandoro committed:
Fix the promise

tnyandoro committed:
Resolve the mixed content error

tnyandoro commented:
This is a Web app that displays weather for different cities in the world per request. 
Removed comments from index.js and weather.js files. 
 Removed  index.js and weather.js files both have similar code. Remove one file since it's redundant.
 Resolved error handling to the logic of the promise.
 Updated readme instructions. 
 PR more descriptive

Forison reviewed (changes requested):
# Status::Changes Required ❌

# 👋 team

#### Good job on addressing some of the issues raised in the previous review, duplicate file was removed and adding error handling promises🤗. However:

* [ ] We still have commented code in the `index.js` file, let us remove all commented code, as requested in the previous review.

* [ ] Let us add a toggle feature, ideally a button that would enable users of the app view temperature in degree Celsius or Fahrenheit when clicked as requested in the previous review.

* [ ] I could not launch the application on my local machine by following the steps outlined in the `readme`, let make sure the steps in the `readme` to get the app to start is really what will cause the application to be launched in the local environment. What happens after the command `npm run build`?

#### That is it, kindly make the changes and submit a re-review😎

Happy coding

Addo Forison

Forison commented on dist/index.html:
```diff
@@ -0,0 +1,44 @@
+    <link rel="stylesheet" href="./style.css" />
+
+    <!-- Bootstrap -->
+    <!-- <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-eOJMYsd53ii+scO/bJGFsiCZc+5NDVN2yr8+0RDqr0Ql0h+rP48ckxlpbzKgwra6" crossorigin="anonymous"> -->
```
remove this line if not needed

Forison commented on src/index.js:
```diff
@@ -0,0 +1,49 @@
+    const { icon, description } = data.weather[0];
+    const { temp, humidity } = data.main;
+    const { speed } = data.wind;
+    //  const { all } = data.clouds;
```
Remove commented code

tnyandoro committed:
Add the f option

tnyandoro committed:
Add the fareignheight option

MuriloRoque reviewed (approved):
## Status: Approved :green_circle: 

### Hello @tnyandoro :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[![Twitter Follow](https://img.shields.io/twitter/follow/MuriloRoquePai1?label=Follow%20Murilo%20on%20Twitter&style=social)](https://twitter.com/MuriloRoquePai1)

tnyandoro commented:
Final commit
